src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -292,32 +737,329 @@ private boolean lineBelongsToSingleCountry(final Line line, final List<Geometry>
         return slices.size() == 1 || CountryBoundaryMap.isSameCountry(slices);
     }
 
+    /**
+     * Marks a line for deletion, if that line is being removed as a relation member. This is safe
+     * to do when the line member was merged with another member to maintain a valid relation and
+     * the merged line is not part of any other relation.
+     *
+     * @param line
+     *            The relation member {@link Line}
+     * @param relationIdentifier
+     *            The parent {@link Relation} identifier
+     */
+    private void markRemovedMemberLineForDeletion(final Line line, final long relationIdentifier)
+    {
+        // Only mark this line for deletion if this line isn't part of any other relations
+        final boolean isPartOfOtherRelations = line.relations().stream()
+                .filter(partOf -> partOf.getIdentifier() != relationIdentifier).findFirst()
```

Comment:
anyMatch might be easier than a combination of findFirst and isPresent

Reply:
Good call!